hazelcast/src/main/java/com/hazelcast/sql/impl/exec/io/flowcontrol/simple/SimpleFlowControl.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec.io.flowcontrol.simple;
+
+import com.hazelcast.sql.HazelcastSqlException;
+import com.hazelcast.sql.impl.QueryId;
+import com.hazelcast.sql.impl.exec.io.flowcontrol.FlowControl;
+import com.hazelcast.sql.impl.operation.QueryFlowControlExchangeOperation;
+import com.hazelcast.sql.impl.operation.QueryOperationHandler;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * Simple implementation of a flow control. The flow control message is sent when the remote end thinks that local end is low
+ * on memory, while this is no longer the case for the local end.
+ */
+public class SimpleFlowControl implements FlowControl {
+    /** Constant LWM threashold. */
+    static final double LWM_PERCENTAGE = 0.25f;
```

Comment:
It took me a couple of minutes to decipher that "LWM" means "low water mark". I'm not strongly against abbreviations/contractions: for instance, while implementing some algorithm or data structure it's better to keep the established terminology used in the original source. But I'm not a big fan of them because with IDEs it takes the same time to type the full versions as the shortened ones. It might make sense to introduce abbreviations somewhere in class/method javadoc before using them.

Reply:
Renamed to "threshold".

Reply:
There is still a lot of LWM mentions in the code.

Reply:
Removed remaining.